android/app/src/main/java/org/fossasia/openevent/fragments/AboutFragment.java

```diff
@@ -110,6 +121,33 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         setUpBookmarksRecyclerView();
         setUpSocialLinksRecyclerView();
 
+        eventLoc.setOnClickListener(v -> {
+            SharedPreferencesUtil.putBoolean(ConstantStrings.IS_MAP_FRAGMENT_FROM_MAIN_ACTIVITY, true);
+            Bundle bundle = new Bundle();
+            bundle.putBoolean(ConstantStrings.IS_MAP_FRAGMENT_FROM_MAIN_ACTIVITY, true);
+            bundle.putString(ConstantStrings.LOCATION_NAME, event.getLocationName());
+
+            FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
+            Context context = getActivity();
+            Fragment mapFragment = ((OpenEventApp) context.getApplicationContext())
+                    .getMapModuleFactory()
+                    .provideMapModule()
+                    .provideMapFragment();
+            mapFragment.setArguments(bundle);
+            fragmentTransaction.replace(R.id.content_frame, mapFragment, FRAGMENT_TAG_REST).commit();
+            ((MainActivity) getActivity()).getSupportActionBar().setTitle(event.getLocationName());
+
+        });
+
+        eventDate.setOnClickListener(v -> {
+            Intent intent = new Intent(Intent.ACTION_INSERT);
+            intent.setType("vnd.android.cursor.item/event");
+            intent.putExtra(CalendarContract.Events.TITLE, event.getName());
+            intent.putExtra(CalendarContract.Events.DESCRIPTION, event.getDescription());
+            intent.putExtra(CalendarContract.EXTRA_EVENT_BEGIN_TIME, DateConverter.formatDateWithDefault(DateConverter.FORMAT_24H, event.getStartsAt()));
+            intent.putExtra(CalendarContract.EXTRA_EVENT_END_TIME, DateConverter.formatDateWithDefault(DateConverter.FORMAT_24H, event.getEndsAt()));
+            startActivity(intent);
+        });
```

Comment:
The data flow is off. Why do we need to persist data in sharedpreferences to communicate between app components? Besides, this can moved to a util class

Reply:
@iamareebjamal  data no longer persists in sharedPreference as it gets cleared and I have moved that calendar code to the Util class. Please review